03/modules/IPDisplay.js

```diff
@@ -0,0 +1,51 @@
+class IPDisplay {
+    constructor(api) {
+        this.api = api;
+        this.sectionElements = document.querySelector('section').children;
+    }
+
+    load() {
+        this.showStatus();
+
+        this.api.loadData()
+            .then(data => {
+                this.showStatus(data);
+
+                return data;
+            })
+            .then(data => {
+                this.insert(data);
+            })
+            .catch(err => this.insert(err.message))
+    }
+
+    showStatus(data) {
+        const [, span] = this.sectionElements;
+
+        if (data) {
+            this.renderTextInfo(span, '0.0.0.0');
+        }
+        else {
+            this.renderTextInfo(span, 'wait...');
+        }
+    }
+
+    insert(ip) {
+        const [button, span] = this.sectionElements;
+
+        button.addEventListener('click', e => {
+            const targetEl = e.target;
+
+            if (targetEl.tagName === 'BUTTON') {
+                this.renderTextInfo(span, ip);
+            }
+        })
+    }
```

Comment:
Raczej `insert()` to nie podpinanie nasłuchiwania. Nazwa metody nic o tym nie mówi.  
Ja bym podpięcie nasłuchiwania zrobił w konstruktorze  lub w metodzie `init()`, która jest w nim uruchamiana.
Ponieważ nasłuchiwanie eventu to część inicjalizacji to powinno być gotowe przed innymi czynnościami.

Reply:
Racja, mało czytelne przez to było. Dodałem poprawkę, która dodatkowo zmieniła funkcjonowanie programu na "bardziej poprawne" tj. dane ładowane są po kliknięciu w przycisk, a co za tym idzie komunikat "wait" wyświetla się za każdym razem w momencie kliknięcia buttona, a nie od razu po załadowaniu strony. 
Dzięki!